src/utils.rs

```diff
@@ -227,3 +203,12 @@ macro_rules! spawn {
         ctx.spawn_local_with_priority($priority, $future);
     };
 }
+
+/// Run a future on the default `MainContext` and block until finished
+#[macro_export]
+macro_rules! block_on {
+    ($future:expr) => {
+        let ctx = glib::MainContext::default();
+        ctx.block_on($future);
+    };
+}
```

Comment:
Hmm. I see the 2 problems with this. The first is that in this macro you use `glib::MainContext::default()`. But I think it's better to use a fully qualified name `gtk::glib::MainContext::default()`. In this way, the calling code does not need to import `glib`.

The second problem is more fundamental. I'm wondering why to use a macro here at all. A function would totally be sufficient.

```suggestion
/// Run a future on the default `MainContext` and block until finished
#[macro_export]
pub(crate) fn block_on<O, F: Future<Output = O>>(fut: F) -> O {
    let ctx = gtk::glib::MainContext::default();
    ctx.block_on(fut)
}
```

Reply:
> A function would totally be sufficient.

Yeah, good point. Both this macro and the other ones can totally be regular functions, I'll convert them.